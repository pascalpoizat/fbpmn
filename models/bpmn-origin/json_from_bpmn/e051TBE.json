{
    "isInterrupt": {
        "BoundaryEvent_0r0bcez": true
    },
    "catN": {
        "StartEvent_1": "NoneStartEvent",
        "Task_1s6tj7r": "AbstractTask",
        "Process_1": "Process",
        "BoundaryEvent_0r0bcez": "TimerBoundaryEvent",
        "EndEvent_14srvev": "TerminateEndEvent",
        "EndEvent_08uegvd": "NoneEndEvent"
    },
    "catE": {
        "SequenceFlow_0yeopwj": "NormalSequenceFlow",
        "SequenceFlow_1co9zfu": "NormalSequenceFlow",
        "SequenceFlow_1131nmm": "NormalSequenceFlow"
    },
    "containN": {
        "Process_1": [
            "StartEvent_1",
            "Task_1s6tj7r",
            "BoundaryEvent_0r0bcez",
            "EndEvent_08uegvd",
            "EndEvent_14srvev"
        ]
    },
    "messageE": {},
    "timeInformation": {
        "BoundaryEvent_0r0bcez": {
            "timerDefinitionValue": null,
            "timerDefinitionType": null
        }
    },
    "targetE": {
        "SequenceFlow_0yeopwj": "Task_1s6tj7r",
        "SequenceFlow_1co9zfu": "EndEvent_08uegvd",
        "SequenceFlow_1131nmm": "EndEvent_14srvev"
    },
    "name": "e051TBE",
    "attached": {
        "BoundaryEvent_0r0bcez": "Task_1s6tj7r"
    },
    "containE": {
        "Process_1": [
            "SequenceFlow_0yeopwj",
            "SequenceFlow_1131nmm",
            "SequenceFlow_1co9zfu"
        ]
    },
    "edges": [
        "SequenceFlow_0yeopwj",
        "SequenceFlow_1131nmm",
        "SequenceFlow_1co9zfu"
    ],
    "messages": [],
    "sourceE": {
        "SequenceFlow_0yeopwj": "StartEvent_1",
        "SequenceFlow_1co9zfu": "BoundaryEvent_0r0bcez",
        "SequenceFlow_1131nmm": "Task_1s6tj7r"
    },
    "nodes": [
        "Process_1",
        "StartEvent_1",
        "Task_1s6tj7r",
        "BoundaryEvent_0r0bcez",
        "EndEvent_08uegvd",
        "EndEvent_14srvev"
    ],
    "nameN": {}
}