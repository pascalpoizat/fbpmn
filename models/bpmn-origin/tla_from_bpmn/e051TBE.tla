---------------- MODULE e051TBE ----------------

EXTENDS TLC, PWSTypes

VARIABLES nodemarks, edgemarks, net

Interest ==
  "Process_1" :> {  }

ContainRel ==
  "Process_1" :> { "StartEvent_1", "Task_1s6tj7r", "BoundaryEvent_0r0bcez", "EndEvent_08uegvd", "EndEvent_14srvev" }

Node == { "Process_1", "StartEvent_1", "Task_1s6tj7r", "BoundaryEvent_0r0bcez", "EndEvent_08uegvd", "EndEvent_14srvev" }

Edge == { "SequenceFlow_0yeopwj", "SequenceFlow_1131nmm", "SequenceFlow_1co9zfu" }

Message == {  }

msgtype ==
  [ i \in {} |-> {}]


source ==
   "SequenceFlow_0yeopwj" :> "StartEvent_1"
@@ "SequenceFlow_1131nmm" :> "Task_1s6tj7r"
@@ "SequenceFlow_1co9zfu" :> "BoundaryEvent_0r0bcez"

target ==
   "SequenceFlow_0yeopwj" :> "Task_1s6tj7r"
@@ "SequenceFlow_1131nmm" :> "EndEvent_14srvev"
@@ "SequenceFlow_1co9zfu" :> "EndEvent_08uegvd"

CatN ==
   "Process_1" :> Process
@@ "StartEvent_1" :> NoneStartEvent
@@ "Task_1s6tj7r" :> AbstractTask
@@ "BoundaryEvent_0r0bcez" :> TimerBoundaryEvent
@@ "EndEvent_08uegvd" :> NoneEndEvent
@@ "EndEvent_14srvev" :> TerminateEndEvent

CatE ==
   "SequenceFlow_0yeopwj" :> NormalSeqFlow
@@ "SequenceFlow_1131nmm" :> NormalSeqFlow
@@ "SequenceFlow_1co9zfu" :> NormalSeqFlow

LOCAL preEdges ==
  [ i \in {} |-> {}]
PreEdges(n,e) == preEdges[n,e]

PreNodes(n,e) == { target[ee] : ee \in preEdges[n,e] }
          \union { nn \in { source[ee] : ee \in preEdges[n,e] } : CatN[nn] \in { NoneStartEvent, MessageStartEvent } }

BoundaryEvent ==
   "BoundaryEvent_0r0bcez" :> [ attachedTo |-> "Task_1s6tj7r", cancelActivity |-> TRUE ]

WF == INSTANCE PWSWellFormed
ASSUME WF!WellTyped
ASSUME WF!WellFormedness

ConstraintNode == TRUE \* none
ConstraintEdge == TRUE \* none
Constraint == TRUE     \* none
INSTANCE PWSConstraints
INSTANCE UserProperties
INSTANCE PWSSemantics

================================================================

