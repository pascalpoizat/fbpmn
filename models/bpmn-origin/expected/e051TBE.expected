Working in /tmp/e051TBE.FXnd0 with 1 worker(s)
[92mtransformation done[0m
<<"Processes=", 1, "Nodes=", 6, "Gateway=", 0, "SF=", 3, "MF=", 0>>
---------- Network01Bag ----------
[X] Prop01Type
     states=7 trans=7 depth=5
[X] Prop02Safe
     states=7 trans=7 depth=5
[X] Prop03Sound
     states=7 trans=7 depth=5
[X] Prop04MsgSound
     states=7 trans=7 depth=5
---------- Network02FifoPair ----------
[X] Prop01Type
     states=7 trans=7 depth=5
[X] Prop02Safe
     states=7 trans=7 depth=5
[X] Prop03Sound
     states=7 trans=7 depth=5
[X] Prop04MsgSound
     states=7 trans=7 depth=5
---------- Network03Causal ----------
[X] Prop01Type
     states=7 trans=7 depth=5
[X] Prop02Safe
     states=7 trans=7 depth=5
[X] Prop03Sound
     states=7 trans=7 depth=5
[X] Prop04MsgSound
     states=7 trans=7 depth=5
---------- Network04Inbox ----------
[X] Prop01Type
     states=7 trans=7 depth=5
[X] Prop02Safe
     states=7 trans=7 depth=5
[X] Prop03Sound
     states=7 trans=7 depth=5
[X] Prop04MsgSound
     states=7 trans=7 depth=5
---------- Network05Outbox ----------
[X] Prop01Type
     states=7 trans=7 depth=5
[X] Prop02Safe
     states=7 trans=7 depth=5
[X] Prop03Sound
     states=7 trans=7 depth=5
[X] Prop04MsgSound
     states=7 trans=7 depth=5
---------- Network06Fifo ----------
[X] Prop01Type
     states=7 trans=7 depth=5
[X] Prop02Safe
     states=7 trans=7 depth=5
[X] Prop03Sound
     states=7 trans=7 depth=5
[X] Prop04MsgSound
     states=7 trans=7 depth=5
---------- Network07RSC ----------
[X] Prop01Type
     states=7 trans=7 depth=5
[X] Prop02Safe
     states=7 trans=7 depth=5
[X] Prop03Sound
     states=7 trans=7 depth=5
[X] Prop04MsgSound
     states=7 trans=7 depth=5
done.
